### DISCUSSION TITLE
SendMode help.

### DISCUSSION THREAD

**Discussion by pieroluha** [autokey/autokey#863] [Q&A] [answered]
How does one use SendMode? 
Sorry, not a python user.

My code below:
```python
# Enter script code
import time
import random

choices = ["something", "something else", "a third thing"]


keyboard.send_key("<enter>")

time.sleep(0.1)
rn = random.randint(0, 2)

time.sleep(0.1)
clipboard.fill_clipboard(choices[rn])

time.sleep(0.1)
text = str(clipboard.get_clipboard())

keyboard.send_keys(text, keyboard.SendMode.CB_CTRL_V)
keyboard.send_key("<enter>")
```

Gives me `AttributeError: 'Keyboard' object has no attribute 'SendMode'`

**Comment by josephj11** [comment, marked as answer]
Welcome to the AutoKey community, @pieroluha !

First, take a look at our [wiki](https://github.com/autokey/autokey/wiki). It has a number of useful articles and lots of examples. Start with [this](https://github.com/autokey/autokey/wiki/Emitting-Keyboard-Events).

We only have one version of our [documentation](https://autokey.github.io/index.html). It refers to AutoKey 0.96.0 and you probably have 0.95.10 which didn't have that option.

Other than that, your script is very close to right.
The code below should work, but I haven't tested it, so it may have some typos...

```
## you don't have to import time, but it doesn't hurt if you do
import random

choices = ["something", "something else", "a third thing"]
keyboard.send_keys("<enter>")  ## don't use send_key() unless you have to.
rn = random.randint(0, 2)

time.sleep(0.1)
clipboard.fill_clipboard(choices[rn]) ## Put your choice directly into the clipboard

time.sleep(0.1)
keyboard.send_keys("<ctrl>+v") ##Paste directly from the clipboard
time.sleep(0.1)  ## just to be safe - probably unnecessary
keyboard.send_keys("<enter>")
```

Pasting from the clipboard should always be your first choice unless you need to send "active" keys that do something on their own like moving the cursor (arrow keys...).

Consider upgrading to [0.96.0](https://github.com/autokey/autokey/releases/). It has a lot of bug fixes and a few new features. [Here's how](https://github.com/autokey/autokey/wiki/Installing).

**Reply by pieroluha** [reply]
Oof. I forgot to check my version...
Many thanks. It works perfectly... 
I don't even know why I was bothering with SendMode...
